README.md

```diff
@@ -10,42 +10,146 @@ data. This file will explain best practices for getting started, building, runni
+   ```bash
+   mvn clean install
+   ```
+
+5. Run the Komet application with the following command:
+   ```bash
+   mvn -f application javafx:run
+   ```
+6. You can open Komet code using your favorite IDE like _Eclipse_ or _IntelliJ Idea_ and try running it from there. 
+While running Komet UI from IDE, you many have to add the follwoing vm argument: 
+   ```
+   -Xmx10g --add-exports javafx.controls/com.sun.javafx.scene.control.behavior=dev.ikm.komet.navigator
+   ```
+ 
+## Usage Examples:
+This section details on the basic design methodology used for developing nex-gen Komet UI.
+1. Komet UI application is moving towards the nex-gen implementation which follows Model-View-View-Model (MVVM) design pattern.
+2. Komet application design is event-based where the subscriber to an event listens for a particular event and when it is triggered,
+desired logic can be executed in the listener code.
+Example:
+   ```java
+   import java.util.UUID;
+   
+   public class MyController {
+      private EvtBus eventBus;
+      private Subscriber<MyDefienedEvent> someMyDefinedEventSubscriber;
+   
+      public void initialize() {
+         someMyDefinedEventSubscriber = evt -> {
+            // Some logic to process the event.
+            if (evt.getEventType() == MyDefienedEvent.SOME_EVENT_1) {
+               // do something.
+            } else if (evt.getEventType() == MyDefienedEvent.SOME_EVENT_2) {
+               //do something else.
+            }
+         };
+         eventBus.subscribe(myTopic, MyDefienedEvent.class, someMyDefinedEventSubscriber);
+      }
+   }
+   
+   public class MyDefienedEvent extends Evt {
+      public static final EvtType<MyDefienedEvent> SOME_EVENT_1 = new EvtType<>(Evt.ANY, "SOME_EVENT_1");
+      public static final EvtType<MyDefienedEvent> SOME_EVENT_2 = new EvtType<>(Evt.ANY, "SOME_EVENT_2");
+   
+      /**
+       * Constructs a prototypical Event.
+       * You can optionally pass arguments in this constructor and set the value as final in the constructor.
+       * The value can be retrived using the getter method for that variable.
+       * @param source         the object on which the Event initially occurred
+       * @param eventType
+       */
+      public MyDefienedEvent(Object source, EvtType eventType) {
+         super(source, eventType);
+      }
+   }
+   
+   public class MySomeClass {
+      private EvtBus eventBus;
+      private UUID someTopic;
+      
+      public MySomeClass(UUID someTopic){
+          this.someTopic = someTopic;
+      }
+      
+      public void someMethod() {
+         eventBus.publish(someTopic, new MyDefienedEvent(this, MyDefienedEvent.SOME_EVENT_1));
+      }
+   }
+   
+   public class MainClass {
+      public static void main(String[] args) {
+         MySomeClass mySomeClass = new MySomeClass(UUID.randomUUID());
+      }
+   }
+   ```
+4. Komet design also cognitive framework to implement MVVM architecture framework.
```

Comment:
Can this read "Komet's design also includes the cognitive..."

Reply:
done